server.js

```diff
@@ -1,92 +1,122 @@
-        let trimmedData = data.props.listingsMap;
-        const numKeys = Object.keys(trimmedData);
-        const numObjects = numKeys.length;
-          if (numObjects > 50) {
-          for (let i = 50; i < numObjects; i++) {
-            delete trimmedData[numKeys[i]];
-          }
-        }
-        for (const id in trimmedData) {
-         if (trimmedData.hasOwnProperty(id)) {
-           delete trimmedData[id].id;
-           delete trimmedData[id].listingType;
-           delete trimmedData[id].listingModel.skeletonImages;
-           delete trimmedData[id].listingModel.auction;
-           delete trimmedData[id].listingModel.branding;
-           delete trimmedData[id].listingModel.brandingAppearance;
-           delete trimmedData[id].listingModel.displaySearchPriceRange;
-           delete trimmedData[id].listingModel.enableSingleLineAddress;
-           delete trimmedData[id].listingModel.hasVideo;
-           delete trimmedData[id].listingModel.promoType;
-           delete trimmedData[id].listingModel.tags;
-         }
-        }
-        const imgParam = req.url.match('[?&]images=([^&]+)');
-        if (imgParam) {
-          const numImages = parseInt(imgParam[1]);
-          if(numImages > 0){
-            for (let key in trimmedData) {
-              if (trimmedData.hasOwnProperty(key)) {
-                const images = trimmedData[key].listingModel.images;
-                if (Array.isArray(images) && images.length > numImages) {
-                  trimmedData[key].listingModel.images = images.slice(0, numImages);
-                }
-              }
-            }
-          }
-          else{
-            for (let key in trimmedData) {
-              if (trimmedData.hasOwnProperty(key)) {
-                delete trimmedData[key].listingModel.images;
-              }
-            }
-          }
-        }
-        else{
-            for (let key in trimmedData) {
-              if (trimmedData.hasOwnProperty(key)) {
-                delete trimmedData[key].listingModel.images;
-              }
-            }
-        }
-        trimmedData = {
-          'totalPages': data.props.pageViewMetadata.searchResponse.SearchResults.totalPages,
-          'page': data.props.pageViewMetadata.searchResponse.SearchResults.page,
-          'totalResults': data.props.pageViewMetadata.searchResponse.SearchResults.totalResults,
-          'resultsPerPage': data.props.listingSearchResultIds.length,
-          'listings': trimmedData
-        };
-        return JSON.stringify(trimmedData);
-      }
-      else{
-         return proxyResData;
-      }
-    }
-    else{
-       return proxyResData;
-    }
-  }
-}));
-
-const server = app.listen(443);
\ No newline at end of file
+#!/usr/bin/env node
+
+const dotenv = require("dotenv");
+const express = require("express");
+const proxy = require("express-http-proxy");
+const cors = require("cors");
+
+// Load environment variables
+const config = dotenv.config();
```

Comment:
@Tophness Whoops.. this line is a bug:

```diff
- const config = dotenv.config();
+ const config = dotenv.config().parsed;
```

Reply:
Another tweak to this improvement if you want to set defaults when no `.env` is committed:

```js
const config = dotenv.config().parsed || {
  DEBUG: 0,
  BIND_HOST: "127.0.0.1",
  BIND_PORT: 3333,
  PROXY_HOST: "https://www.domain.com.au",
};
config.DEBUG = ["1", "true"].includes(config.DEBUG) || false;
```

Alternatively, checking if `config` is `falsy` before the `config.DEBUG` line and throwing an error informing the user that they need to setup the `.env` properly.

Reply:
Let's just be super stock standard about it...

```diff
diff --git a/server.js b/server.js
index 40d2f24..804cec9 100755
--- a/server.js
+++ b/server.js
@@ -6,7 +6,14 @@ const proxy = require("express-http-proxy");
 const cors = require("cors");
 
 // Load environment variables
-const config = dotenv.config();
+dotenv.config();
+
+const config = process.env;
+
+if (!config.BIND_HOST) throw new Error("required env var BIND_HOST not set");
+if (!config.BIND_PORT) throw new Error("required env var BIND_PORT not set");
+if (!config.PROXY_HOST) throw new Error("required env var PROXY_HOST not set");
+
 config.DEBUG = ["1", "true"].includes(config.DEBUG) || false;
 
 const app = express();
```